### ISSUE TITLE
[LINT_BUG]: re-exported objects are flagged as unused

### ISSUE THREAD

**Issue by caldwellst** [Appsilon/box.linters#152]
### box.linters version

`0.10.5`

### Sample source code to lint

# src/a.R
```r
#' @export
a <- function(x) sum(x)
```

# src/__init__.R
```r
#' @export
box::use(src/a[...])
```

# .Rprofile
```r
options(box.path = getwd())
```

### Lint command used

lintr::lint_dir(
  linters = lintr::linters_with_defaults(defaults = box.linters::box_default_linters)
)

### Lint result

```r
[box_unused_attached_mod_linter] Three-dots attached module unused.
```

### Expected result

Should successfully lint, as the use of `__init__.R` files to re-export objects from scripts at the module level is part of the `{box}` paradigm.
